test/mpi/test_allgatherv.cc

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "test_mpi.h"
+#include "mpi_util.h"
+
+#define TEST_DT UCC_DT_UINT32
+
+TestAllgatherv::TestAllgatherv(size_t _msgsize, ucc_test_mpi_inplace_t _inplace,
+                               ucc_memory_type_t _mt, ucc_test_team_t &_team) :
+    TestCase(_team, _mt, _msgsize, _inplace)
+{
+    size_t dt_size = ucc_dt_size(TEST_DT);
+    size_t count = _msgsize/dt_size;
+    int rank, size;
+    MPI_Comm_rank(team.comm, &rank);
+    MPI_Comm_size(team.comm, &size);
+
+    UCC_CHECK(ucc_mc_alloc((void**)&counts, size * sizeof(uint32_t),
+                           UCC_MEMORY_TYPE_HOST));
+    UCC_CHECK(ucc_mc_alloc((void**)&displacements, size * sizeof(uint32_t),
+                           UCC_MEMORY_TYPE_HOST));
+    UCC_CHECK(ucc_mc_alloc(&rbuf, _msgsize*size, _mt));
+    UCC_CHECK(ucc_mc_alloc(&check_buf, _msgsize*size, _mt));
+    for (int i = 0; i < size; i++) {
+        counts[i] = count;
+        displacements[i] = i * count;
+    }
+    if (TEST_NO_INPLACE == inplace) {
+        args.mask = 0;
+        UCC_CHECK(ucc_mc_alloc(&sbuf, _msgsize, _mt));
+        init_buffer(sbuf, count, TEST_DT, _mt, rank);
+    } else {
+        args.mask = UCC_COLL_ARGS_FIELD_FLAGS;
+        args.flags = UCC_COLL_ARGS_FLAG_IN_PLACE;
+        init_buffer((void*)((ptrdiff_t)rbuf + rank*count*dt_size),
+                    count, TEST_DT, _mt, rank);
+        init_buffer((void*)((ptrdiff_t)check_buf + rank*count*dt_size),
+                    count, TEST_DT, _mt, rank);
+    }
+    args.coll_type                = UCC_COLL_TYPE_ALLGATHERV;
+    args.src.info.buffer          = sbuf;
```

Comment:
Now that i think of it: we probably should only init "src" here in the test in the NO_INPLACE case only, leaving garbage or NULLs otherwise. In this case, the test would also stress our implementation that it does not use incorrect fields. wdyt?

Reply:
sure, sounds good